doc/introduction/unsupported_gradients.rst

```diff
@@ -96,8 +96,8 @@ Adjoint differentiation
 
 PennyLane implements the adjoint differentiation method from
 `2009.02823 <https://arxiv.org/pdf/2009.02823.pdf>`__, which only discusses
-the gradient of expectation of observables. The implementation is specific to the paper, hence the return statement
-of the quantum function wrapped in ``qml.qnode`` can only contain :func:`~.pennylane.expval` as a measurement.
+the gradient of expectation of observables. The implementation is specific to the paper, hence the best performance
+will occur when only using expectation values of observables.

```

Comment:
I think this is outdated if we have support for VJPs with adjoint.

If we are supporting VJPs with adjoint, then adjoint should be efficient _as long as the final cost function that is being differentiated is a scalar_.

So a QNode with a single expectation value, _or_ a statevector/probability vector/multiple expvals that is being reduced to a scalar, etc.

This is why `<H>` is efficient, because we can reduce multiple expectation values to a scalar output.

Reply:
I have updated the docs to recommend using `device_vjp=True` when differentiating non-expectation values.

The challenge is *should be efficient* versus *is efficient*. I don't yet have enough numbers to promise that it does have decent performance scaling.

I have tried one case, and adjoint+device vjp seems reasonably decent for that one case, but I don't think that's enough evidence to recommend it across the board.

Reply:
@albi3ro in this case though, using VJPs with backpropagation-style methods (like adjoint) will provably be better performant if the cost function being differentiated has a scalar output, compared to not using VJPs.

I understand we haven't done more formal benchmarking here, but since we are only promising 'best performance' (not efficient performance scaling, which is more rigorous), I would say that is fine. E.g., the following should be okay here:

```
The implementation is specific to the paper, hence the best performance
will occur only when the final cost function being differentiated has a
scalar value, and statevectors are not returned from QNodes.
```

Is there an underlying reason for the performance concern?

Reply:
I've changed the documentation here a little bit, so hopefully it's better now.

I've been running some more numbers, and getting more comfortable with recommending using adjoint diff+state, as long as `device_vjp=True`. 

My reasons for concern come back to historically seeing the difference between "should be" and "is" when it comes to performance, as implementation details can slow everything down or overheads can dominate. For example, when we starting binding device vjp's, they were substantially slower then the full jacobian path. And I have seen that the full jacobian for adjoint+state is very memory-hungry.

![scaling](https://github.com/PennyLaneAI/pennylane/assets/6364575/b1fc0a3a-28c5-429a-a96f-c1a873902b48)